src/write.rs

```diff
@@ -149,38 +149,75 @@ struct ZipRawValues {
     compressed_size: u64,
     uncompressed_size: u64,
 }
-
+trait Sealed {}
+/// File options Extensions
+#[allow(unknown_lints)]
+#[allow(private_bounds)]
+#[doc(hidden)]
+pub trait FileOptionExtension: Default + Sealed {
```

Comment:
This turns out not to be allowed before 1.74.0, and the current MSRV is 1.70.0. To fix this, make Sealed a public member of a private module; see https://predr.ag/blog/definitive-guide-to-sealed-traits-in-rust/#sealing-traits-with-a-supertrait and https://doc.rust-lang.org/error_codes/E0445.html#.